src/twisted/mail/imap4.py

```diff
@@ -5530,21 +5534,22 @@ def decoder(s, errors=None):
     """
     r = []
     decode = []
+    s = memory_cast(s, 'c')
     for c in s:
-        if c == '&' and not decode:
+        if c == b'&' and not decode:
             decode.append('&')
-        elif c == '-' and decode:
+        elif c == b'-' and decode:
             if len(decode) == 1:
                 r.append('&')
             else:
-                r.append(modified_unbase64(''.join(decode[1:])))
+                r.append(modified_unbase64(b''.join(decode[1:])))
             decode = []
         elif decode:
             decode.append(c)
         else:
-            r.append(c)
+            r.append(c.decode())
     if decode:
-        r.append(modified_unbase64(''.join(decode[1:])))
+        r.append(modified_unbase64(b''.join(decode[1:])))
     return (''.join(r), len(s))
```

Comment:
Sure, I already did a [similar change](https://github.com/berdario/twisted/commit/e00024bd680f0ac44ce710acd5fc5c65c7e67f27) in the past, but I was not sure if it would just add noise